provisioning/baremetal_config.go

```diff
@@ -142,20 +143,14 @@ func getProvisioningIPCIDR(config *metal3iov1alpha1.ProvisioningSpec) *string {
 	return nil
 }
 
-func getDeployKernelUrl(config *metal3iov1alpha1.ProvisioningSpec) *string {
-	if config.ProvisioningIP != "" {
-		deployKernelUrl := fmt.Sprintf("http://%s/%s", net.JoinHostPort(config.ProvisioningIP, baremetalHttpPort), baremetalKernelUrlSubPath)
-		return &deployKernelUrl
-	}
-	return nil
+func getDeployKernelUrl() *string {
+	deployKernelUrl := fmt.Sprintf("http://%s/%s", net.JoinHostPort("api", strconv.Itoa(imageCachePort)), baremetalKernelUrlSubPath)
```

Comment:
Should we always access these over the API VIP? I think users would expect that if there's a provisioning network, the traffic would happen over it.

Reply:
It could still hit the image cache port. I'm just imagining that these IPA images are a few hundred megs, and a rapid scale out of a few dozen machines is going to use a lot of bandwidth on the machine networks.

Reply:
Hmm, I was working on the theory that this URL was used for traffic only between Ironic and httpd, so all internal to the box anyway. If Ironic is passing this URLs to BMCs then it definitely needs to use the provisioning IP when it is available.

Reply:
The API VIP isn't guaranteed to be on the same box as Ironic, though.

Reply:
Actually if that's the case then this won't work at all, and we'll need to put an IP address always, since the DNS is only resolvable by ironic.
And not even that, judging by the failure in the gate.

Reply:
So could we use the ProvisioningIP when we have one, and if not use the API VIP?

Reply:
The DNS wasn't the problem, but actually we could just always use localhost:6181. Then it would always hit the local image cache even when there's no provisioning VIP, so no E-W traffic in any configuration. And as a bonus, it will likely work without waiting for BMO fixes because the image cache comes up more or less instantly on the host with the metal3 Pod.

Reply:
Ok I think this makes sense then. PXE clients get the URL from the ram disk from the TFTP server. 

Localhost should probably be ok here